src/spikeinterface/core/template.py

```diff
@@ -0,0 +1,197 @@
+import numpy as np
+import json
+from dataclasses import dataclass, field, astuple
+from .sparsity import ChannelSparsity
+
+
+@dataclass(kw_only=True)
+class Templates:
+    """
+    A class to represent spike templates, which can be either dense or sparse.
+
+    Attributes
+    ----------
+    templates_array : np.ndarray
+        Array containing the templates data.
+    sampling_frequency : float
+        Sampling frequency of the templates.
+    nbefore : int
+        Number of samples before the spike peak.
+    sparsity_mask : np.ndarray, optional
+        Binary array indicating the sparsity pattern of the templates.
+        If `None`, the templates are considered dense.
```

Comment:
Just a question here @h-mayorquin. Binary array means 1 and 0s and boolean array would mean True/False. I don't know what the term would be for something that can either be 1/0 or True/False. But are you hoping that this accepts the format 1/0. (Maybe this is just super annoying semantics, but I'm curious how you think about this?)

Reply:
No, this should be boolean :)